src/mpid/ch4/netmod/ucx/ucx_types.h

```diff
@@ -18,9 +18,18 @@
 #define UCP_PEER_NAME_MAX         HOST_NAME_MAX
 
 /* Active Message Stuff */
-#define MPIDI_UCX_NUM_AM_BUFFERS       (64)
 #define MPIDI_UCX_MAX_AM_EAGER_SZ      (16*1024)
-#define MPIDI_UCX_AM_TAG               (1ULL << MPIR_Process.tag_bits)
+#define MPIDI_UCX_AM_BUF_COUNT         8
+#define MPIDI_UCX_AM_BUF_SIZE          MPIDI_UCX_MAX_AM_EAGER_SZ
```

Comment:
Do we need this one? We can just we `MPIDI_UCX_MAX_AM_EAGER_SZ` for the am buffer allocation and the checking on packing types.

Reply:
`BUF_SIZE` conceptually can be different from `EAGER_SZ`. The latter is a protocol decision and could be controlled by the user (at some point the code may evolve into that). Besides, the buffer may need account for extra meta data. Having separate definitions allows for the later flexibility.

Reply:
It is the ucx library that controls the actual eager limit. The EAGER size is just a limit on how much data we will send on one ucx_tag_send. I cannot think of any benefit of having a AM_BUF_SZ not equal to EAGER_SZ.

Reply:
I think there is benefit at least in cleaner semantics. When we allocating a buffer of `EAGER_SZ`, then we need understand the reason, which requires the knowledge of the purpose of `EAGER_SZ` and how it is used. In fact, I think `EAGER_SZ` is a derivative concept from `BUF_SIZE`. So if anything, from ucx point of view, we probably don't need `MPIDI_UCX_MAX_AM_EAGER_SZ`, but I guess at some context such as `MPIDI_NM_am_eager_limit`, it reads better having the explicit macro. Does it bother you so much that we have two separate macros having the same value? Their connection is clearly defined here.

Reply:
I see. I think it is confusing to have two macros. It makes more sense to get rid of the AM_EAGER_SZ and derive the eager limit from AM_BUF_SZ. We did the same in POSIX---eager limit is derived from CELL SIZE.

Reply:
I think I can agree with that.